lib/ts/framework/utils.ts

```diff
@@ -23,7 +22,72 @@ import type { HTTPMethod } from "../types";
 import { COOKIE_HEADER } from "./constants";
 import { getFromObjectCaseInsensitive } from "../utils";
 import contentType from "content-type";
-import inflate from "inflation";
+import pako from "pako";
+import { decompress as brotliDecompress } from "brotli";
+
+async function inflate(stream: IncomingMessage): Promise<string> {
+    if (!stream) {
+        throw new TypeError("argument stream is required");
+    }
+
+    const encoding = (stream.headers && stream.headers["content-encoding"]) || "identity";
+
+    let decompressedData: Uint8Array | string;
+
+    if (encoding === "gzip" || encoding === "deflate") {
+        const inflator = new pako.Inflate();
+
+        for await (const chunk of stream) {
+            inflator.push(chunk, false);
+        }
+
+        if (inflator.err) {
+            throw new Error(`Decompression error: ${inflator.msg}`);
+        }
+
+        decompressedData = inflator.result;
+    } else if (encoding === "br") {
+        const chunks: Buffer[] = [];
+        for await (const chunk of stream) {
+            chunks.push(chunk);
+        }
+        const compressedData = Buffer.concat(chunks);
+        decompressedData = brotliDecompress(compressedData);
+    } else {
+        // Handle identity or unsupported encoding
+        decompressedData = Buffer.concat([]);
+        for await (const chunk of stream) {
+            decompressedData = Buffer.concat([decompressedData, chunk]);
+        }
+    }
+
+    if (typeof decompressedData === "string") return decompressedData;
+
+    return new TextDecoder().decode(decompressedData);
+}
+
+export function parseParams(string: string): object {
+    // Set up a new URLSearchParams object using the string.
+    const params = new URLSearchParams(string);
+
+    // Get an iterator for the URLSearchParams object.
+    const entries = params.entries();
+
+    const result: { [key: string]: any } = {};
+
+    // Loop through the URLSearchParams object and add each key/value
+    for (const [key, value] of entries) {
+        // Split comma-separated values into an array.
+        result[key] = value.split(",");
+
+        // If a key does not have a value, delete it.
+        if (!value) {
```

Comment:
Would this ever be true? I suspect the line above would throw in that case.

Reply:
It is for the case of an empty string. The value would never contain a `null` or `undefined` as it goes through URLSearchParams. I also have a test case for this.

Take the following example:

```js
const p = parseParam("?a=1&b=")

// In this case b's value would evaluate to an empty string and parseParams will remove it
console.log(p).

// p should be { "a": ["1"] }
```

Reply:
I see, my bad. However, that is a very unexpected behaviour.